Lib/urllib/parse.py

```diff
@@ -973,9 +973,9 @@ def splitport(host):
     """splitport('host:port') --> 'host', 'port'."""
     global _portprog
     if _portprog is None:
-        _portprog = re.compile('(.*):([0-9]*)$', re.DOTALL)
+        _portprog = re.compile('(.*):([0-9]*)')
```

Comment:
The same as for `splithost()`. `splithost('example.org\n')` will return a tuple `('example.org\n', None)`.